### PR TITLE
LPD-36908 - Variable shipping rate is calculated on total price even if a non-shippable item is on the list

### PR DESCRIPTION
**Pull request by victorcdp**
@andrea-ale-sbarra Can you review this? Tied to FIRE LPP. Thank you!

https://liferay.atlassian.net/browse/LPD-36908

### CODE REVIEW

**Comment by victorcdp** [comment]
ci:test:sf

**Comment by victorcdp** [comment]
ci:test:playwright

File: modules/apps/commerce/commerce-shipping-engine-fixed-web/src/main/java/com/liferay/commerce/shipping/engine/fixed/web/internal/ByWeightCommerceShippingEngine.java [hunk 1]
```diff
@@ -301,6 +299,20 @@ private List<CommerceShippingOption> _getCommerceShippingOptions(
...
 	}
 
+	private BigDecimal _getOrderShippableSubtotal(CommerceOrder commerceOrder) {
+		BigDecimal subtotal = BigDecimal.ZERO;
+		List<CommerceOrderItem> commerceOrderItems =
+			commerceOrder.getCommerceOrderItems();
+
+		for (CommerceOrderItem commerceOrderItem : commerceOrderItems) {
+			if (commerceOrderItem.isShippable()) {
+				subtotal = subtotal.add(commerceOrderItem.getFinalPrice());
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-shipping-engine-fixed-web/src/main/java/com/liferay/commerce/shipping/engine/fixed/web/internal/ByWeightCommerceShippingEngine.java:309]: I'm not 100% sure you should use this or `getFinalPriceWithTaxAmountMoney()`

I think there is a config that says how to calculate price.

**Review by andrea-ale-sbarra — COMMENTED** [review]
@luca-pellizzon can you have a look at this please?

Thank you!

File: modules/apps/commerce/commerce-shipping-engine-fixed-web/src/main/java/com/liferay/commerce/shipping/engine/fixed/web/internal/ByWeightCommerceShippingEngine.java (hunk 1 above)
Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-shipping-engine-fixed-web/src/main/java/com/liferay/commerce/shipping/engine/fixed/web/internal/ByWeightCommerceShippingEngine.java:309]: I think taxes are applied on top of this, as before the change where we were using the order's subtotal.

**Review by luca-pellizzon — APPROVED** [review]

**Comment by luca-pellizzon** [comment]
ci:test:playwright

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:relevant

**Comment by victorcdp** [comment]
ci:test:playwright

**Comment by luca-pellizzon** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-shipping-engine-fixed-web/src/main/java/com/liferay/commerce/shipping/engine/fixed/web/internal/ByWeightCommerceShippingEngine.java (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-shipping-engine-fixed-web/src/main/java/com/liferay/commerce/shipping/engine/fixed/web/internal/ByWeightCommerceShippingEngine.java:309]: Talked offline with @luca-pellizzon , the solution is right!
Thank you!

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by luca-pellizzon** [comment]
ci:forward

**Pull request closed without merging**